packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -90,15 +90,37 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
             this.driverPolicies.enableRestLess,
             resolvedUrl.endpoints.ordererUrl,
         );
+
+        const documentPostCreateCallback = this.tokenProvider.documentPostCreateCallback;
+
         // the backend responds with the actual document ID associated with the new container.
-        const documentId = await ordererRestWrapper.post<string>(
+        const res = await ordererRestWrapper.post<{ id: string, token?: string } | string>(
             `/documents/${tenantId}`,
             {
                 summary: convertSummaryToCreateNewSummary(appSummary),
                 sequenceNumber: documentAttributes.sequenceNumber,
                 values: quorumValues,
+                generateToken: documentPostCreateCallback !== undefined,
             },
         );
+
+        // For supporting backward compatibility, when the request has generateToken === true, it will return
```

Comment:
I'd highly recommend leaving enough breadcrumbs in the code for next developer to be able to assess if it's now safe to remove back-compat code, and what code needs to be removed / modified. We have plenty of examples where it's not the case and code stays littered with dead code and comments like that.